users/views.py

```diff
@@ -199,23 +204,19 @@ def post(cls, request: HttpRequest, pk: int) -> Union[Response, HttpResponseRedi
         return redirect("custom-user-update-by-admin", pk=pk)
 
 
-class UserList(APIView):
-    serializer_class = UserListSerializer
-    renderer_classes = [renderers.TemplateHTMLRenderer]
+@method_decorator(login_required, name="dispatch")
+class UserList(ListView):
     template_name = "users_list.html"
-    permission_classes = (permissions.IsAuthenticated,)
-
-    @classmethod
-    def get_queryset(cls) -> CustomUser:
-        logger.info("Get user list in queryset")
-        return CustomUser.objects.order_by("id")
-
-    @classmethod
-    def get(cls, request: HttpRequest) -> Response:
-        logger.info(f"User with id: {request.user.pk} get to the UserList view")
-        users_queryset = cls.get_queryset()
-        users_serializer = UserListSerializer(context={"request": request})
-        return Response({"serializer": users_serializer, "users_list": users_queryset})
+    model = CustomUser
+    queryset = CustomUser.objects.prefetch_related("projects")
+
+    def get_context_data(
+        self, *, object_list: Optional[List] = None, **kwargs: Any  # pylint: disable=unused-argument
+    ) -> dict:
+        if self.request.user.is_admin:
```

Comment:
Nice try ;) But lets add a decorator for it that we can reuse all over the place. This decorator should take a list of user types that are allowed for a given view, this way we have only one decorator for everything!